api4/post.go

```diff
@@ -330,26 +330,42 @@ func searchPosts(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	includeDeletedChannels := r.URL.Query().Get("include_deleted_channels") == "true"
+	params := model.SearchParameterFromJson(r.Body)
 
-	props := model.StringInterfaceFromJson(r.Body)
-
-	terms, ok := props["terms"].(string)
-	if !ok || len(terms) == 0 {
+	if params.Terms == nil || len(*params.Terms) == 0 {
 		c.SetInvalidParam("terms")
 		return
 	}
+	terms := *params.Terms
+
+	timeZoneOffset := 0
```

Comment:
This process for parsing the search parameters is a bit complicated for me right now since it turns the body of the request into a `model.SearchParameter`, sets defaults, passes the fields into `App.SearchPostsInTeam`, and then recombines them back into an array of `model.SearchParams`. I don't know if it's something to change now since it'd take some extra time to refactor and work out, but it could definitely be made simpler

Reply:
Ya it's a bit strange, but I only wanted to fix what was needed.